### PR TITLE
Add FeatureGroupSubGroup plugin

### PR DESCRIPTION
**Pull request by shtrom**
This plugin [0] allows can be used to either:
* create nested overlays that can be turned on or off together or separately,

```python
fg = folium.FeatureGroup()
m.add_child(fg)

g1 = folium.plugins.FeatureGroupSubGroup(fg, 'g1')
m.add_child(g1)
g2 = folium.plugins.FeatureGroupSubGroup(fg, 'g2')
m.add_child(g2)

l = folium.LayerControl().add_to(m)
```

* make multiple overlays part of the same MarkerCluster,

```python
mcg = folium.plugins.MarkerCluster(control=False)
m.add_child(mcg)

g1 = folium.plugins.FeatureGroupSubGroup(mcg, 'g1')
m.add_child(g1)
g2 = folium.plugins.FeatureGroupSubGroup(mcg, 'g2')
m.add_child(g2)

l = folium.LayerControl().add_to(m)
```

[0] https://github.com/ghybs/Leaflet.FeatureGroup.SubGroup

Signed-off-by: Olivier Mehani <shtrom@ssji.net>

### CODE REVIEW

**Comment by shtrom** [comment]
@ghybs FYI.

**Comment by Conengmo** [comment]
This looks great, well done. This fits in well I think. Before I merge I have two requests:

- Your docstring is already very thorough. But a lot of our users check the Jupyter Notebook examples to see how stuff works. Could you add an example to [the plugins notebook](https://github.com/python-visualization/folium/blob/master/examples/Plugins.ipynb)?
- Please add a line to the changelog.

**Comment by shtrom** [comment]
I added a test, quickly hacked from what I could gather from the MarkerCluster one, but couldn't work out how to exercise it. Neither running `python setup.py test` nor even running the script manually seems to do the right thing (I didn't get a failure with a purposefully failing `assert`).

File: tests/plugins/test_feature_group_sub_group.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+"""
+Test MarkerCluster
+------------------
+"""
+
+from __future__ import (absolute_import, division, print_function)
+
+import folium
+
+from folium import plugins
```
[the remark this answers is not in the document]
Comment (shtrom) [reply tests/plugins/test_feature_group_sub_group.py]: Fixed.

**Pull request merged**